### PR TITLE
fix: handle deprecated class LNumber in nikic/php-parser v5

### PR DESCRIPTION
**Pull request by nikophil**
I think since the `Automapper` is the only entrypoint of this whole lib, we can add `class_alias()` in this class, and problem would be solved.

WDYT?

### CODE REVIEW

**Comment by joelwurtz** [comment]
Not a fan of this, i would keep declaring the alias where it is really needed, not on the top class, main problem is that if we change this later (like some lib that don't depend on the AutoMapper class) we would not have those alias

**Comment by GromNaN** [comment]
Using `class_alias` for classes that doesn't belong to this library can lead to other issues (like an other library checking of the class exist ...). That's bad that php-parser doesn't provide a clean migration path. The best would be that php-parser 4.x would have provided the new classes directly.

A clean alternative would be to create factory functions here, that uses the correct class name depending on the php-parser version.

**Comment by nikophil** [comment]
> A clean alternative would be to create factory functions here, that uses the correct class name depending on the php-parser version.

this was actually what I done in an earlier version :sweat_smile:

**Review by joelwurtz — APPROVED** [review]

File: src/php-parser.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Scalar;
+
+/**
+ * Constructs an integer number scalar node.
+ *
+ * @param int                  $value      Value of the number
+ * @param array<string, mixed> $attributes Additional attributes
```
Comment (GromNaN) [inline src/php-parser.php:19]: This functions should be `@internal`

**Pull request merged**